### ISSUE TITLE
Migrate from our custom go BUILD rules to the upstream ones.
For the most part, this leaves the custom files and macros in
place to minimize diffs. [phab:D1044]

### ISSUE THREAD

**Issue by kythebot** [google/kythe#1384]
_Created by **shahms** at 2016-09-12 18:21:44_: 

Move to using upstream Go rules

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2016-09-12 19:39:17_:

Please add a comment saying what this is for. It doesn't need to be exhaustive, but a summary would be helpful.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2016-09-12 19:39:17_:

Can you please add some comments to this, summarizing what it's for?

(I realize that expands the diff a bit, but I think it'll help)

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2016-09-12 19:39:17_:

Please comment the ones that are left, even if only briefly.

E.g.
```
# This macro generates a go_library for all the Go files in a single
# directory, and a go_test rule if "_test.go" files are present. This is
# a shortcut rule for single-package directories.
```

or words to that effect.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2016-09-12 19:39:17_:

Odd indentation?

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2016-09-12 20:17:29_:

Where? This indentation is consistent and hasn't changed (AFAICT).

**Comment by kythe** [comment]
_**shahms** wrote at 2016-09-12 20:19:03_:

  - Add comments to existing and new build rules/macros

# Updating (earlier comment): Migrate from our custom go BUILD rules to the upstream ones.
For the most part, this leaves the custom files and macros in
place to minimize diffs.

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2016-09-12 20:21:24_:

nit: s/go/Go/

**Comment by kythe** [comment]
_**fromberger** commented inline (context lost during migration) at 2016-09-12 20:21:24_:

Nevermind. I misread and thought this was supposed to be inside the loop. :)

**Comment by kythe** [comment]
_**fromberger** accepted this revision at 2016-09-12 20:22:08_

**Comment by kythe** [comment]
_**shahms** closed via commit 57552fcf82d43ff0c26d16744822cb674b99fe40 at 2016-09-12 20:22:51_
